### PR TITLE
fix(*): bumps AGP, fixes navigation and theming

### PR DESCRIPTION
**Pull request by carotkut94**
Changes/Fixes in PR
* Fixes navigation issues post login/signup
* Fixes theming for login/signup screen reported on [20](https://github.com/PatilShreyas/NotyKT/issues/120)
* Adds visual transformation for password fields
* bumps AGP

Further more, the issue regarding the committed files such as, `dependencies.gradle` or `noty-android/gradle/wrapper/gradle-wrapper.properties` is because of  different line endings and the I believe the app is first developed/build on a windows machine for more info [Stackoverflow](https://stackoverflow.com/a/26490990/7920783)

Or am I missing something here?

further more I believe a template for PR creation should be put in place with  checkboxex for 
- [x] ran ./gradlew ktlintformat
- [x] ran ./gradlew ktlintCheck

or at least these steps getting added in contributing guideline as well.

### CODE REVIEW

File: noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/LoginScreen.kt [hunk 1]
```diff
@@ -54,28 +56,29 @@ import kotlinx.coroutines.ExperimentalCoroutinesApi
 @Composable
 fun LoginScreen(navController: NavHostController, loginViewModel: LoginViewModel) {
 
-    val viewState = loginViewModel.authFlow.collectAsState(initial = null).value
-
-    when (viewState) {
+    when (val viewState = loginViewModel.authFlow.collectAsState(initial = null).value) {
```
Comment (PatilShreyas) [inline noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/LoginScreen.kt:59]: The earlier version looks more readable. Please revert this change.

File: noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/SignUpScreen.kt [hunk 2]
```diff
@@ -53,15 +55,16 @@ fun SignUpScreen(
     viewModel: RegisterViewModel
 ) {
 
-    val viewState = viewModel.authFlow.collectAsState(initial = null).value
-
-    when (viewState) {
+    when (val viewState = viewModel.authFlow.collectAsState(initial = null).value) {
```
Comment (PatilShreyas) [inline noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/SignUpScreen.kt:58]: Same as above ⬆️

**Review by PatilShreyas — CHANGES_REQUESTED** [review]
@carotkut94 This change looks good to me, thanks for this PR 😃. Just a small comment below 👇🏻

File: noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/SignUpScreen.kt (hunk 2 above)
Comment (carotkut94) [reply noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/SignUpScreen.kt:58]: @PatilShreyas I believe, thats an instance of view state, and its only being used in when scope, and creation of it above when, increases the scope and will allow mutation of state outside  the when statement,  but for now there is no usage of view state outside of it, thus restrictes its mutation, let me know if it feels like a good enough reason to keep it as is, or I can revert it, otherwise.

File: noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/SignUpScreen.kt [hunk 3]
```diff
@@ -53,15 +55,16 @@ fun SignUpScreen(
...
             navController.navigate(
                 route = Screen.Notes.route,
                 builder = {
                     launchSingleTop = true
+                    popUpTo(Screen.SignUp.route) {
+                        inclusive = true
+                    }
                 }
             )
         }
```
Comment (PatilShreyas) [inline noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/SignUpScreen.kt:70]: @carotkut94 I think this isn't working. I tested the built app and produced scenario as follows:

- Clean install application
- Launch app 
- Clicked on "Sign up"
- Entered credentials and clicked "Sign Up"
- (_Notes screen launched_) Press back 🔙 

After these procedures, the Sign Up screen was launched. This needs to be fixed.

**Review by PatilShreyas — CHANGES_REQUESTED** [review]

File: noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/LoginScreen.kt [hunk 4]
```diff
@@ -54,28 +56,29 @@ import kotlinx.coroutines.ExperimentalCoroutinesApi
...
             )
         }
     }
 
-    LazyColumn {
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(Color.White)
+    ) {
```
Comment (PatilShreyas) [inline noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/LoginScreen.kt:79]: This change isn't working I guess. Please look into it.

**Review by PatilShreyas — CHANGES_REQUESTED** [review]

File: noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/SignUpScreen.kt (hunk 2 above)
Comment (PatilShreyas) [reply noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/SignUpScreen.kt:58]: Yeah please revert it. It looks more readable.

**Comment by PatilShreyas** [comment]
Thanks, @carotkut94 for your suggestion regarding contribution checklist. It's a really good suggestion 💯  and I've done the necessary things for this. I've updated contributing guidelines as well as defined PR template 😃.

File: noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/LoginScreen.kt (hunk 4 above)
Comment (carotkut94) [reply noty-android/app/composeapp/src/main/java/dev/shreyaspatil/noty/composeapp/view/screen/LoginScreen.kt:79]: I missed that fix on Signup page, will push the changes for all

**Comment by PatilShreyas** [comment]
@carotkut94 Thanks for your contribution really appreciate it. These fixes have been already implemented by #135 so closing this. Feel free to contribute again!
